modules/stack/main.tf

```
@@ -39,6 +39,66 @@ resource "spacelift_stack" "default" {
   before_plan    = var.before_plan
 
   protect_from_deletion = var.protect_from_deletion
+
+  dynamic "azure_devops" {
+    for_each = var.azure_devops ? [var.azure_devops] : []
```

```suggestion
    for_each = var.azure_devops != null ? [true] : []
```

`var.azure_devops` is a map, not boolean.
Also, it's not necessary to use it again in `[]`, we can use just `[true]` since we just iterate over that one item without actually using it